### ISSUE TITLE
`deno bundle` fails to resolve react/jsx-runtime

### ISSUE THREAD

**Issue by pheuter** [denoland/deno#13389]
I'm seeing an issue with this when trying to do `deno bundle` with `react/jsx-runtime`, created repro here with description of error: https://github.com/pheuter/hello-deno#blocking-issue.

Currently unable to bundle client-side code using `react/jsx-runtime` despite `deno run` working just fine using the same code and config:

```
» deno bundle --config deno.json client/index.tsx static/client.js
Check file:///Users/mark/src/pheuter/hello-deno/client/index.tsx
Bundle file:///Users/mark/src/pheuter/hello-deno/client/index.tsx
error: Unable to output during bundling.

Caused by:
    0: load_transformed failed
    1: failed to analyze module
    2: failed to resolve https://esm.sh/react@17.0.2/jsx-runtime from file:///Users/mark/src/pheuter/hello-deno/client/index.tsx
    3: Cannot resolve "https://esm.sh/react@17.0.2/jsx-runtime" from "file:///Users/mark/src/pheuter/hello-deno/client/index.tsx".
```

```sh
» deno --version
deno 1.17.3 (release, aarch64-apple-darwin)
v8 9.7.106.15
typescript 4.5.2
```

Oddly enough, the issue appears to go away when I explicitly do `/** @jsxImportSource https://esm.sh/react@17.0.2 */` in every tsx file, which I'm trying to avoid with this optimization in the first place.

**Comment by kitsonk** [comment]
We need to inject this module into the bundle when present in the config.  Even though it is in the module graph, because swc does the bundling, it isn't away of it directly.

**Comment by milky2028** [comment]
I also see this issue when doing `deno compile`.

**Comment by lucacasonato** [comment]
Fixed by #15561.

Still needs a test

**Comment by marvinhagemeister** [comment]
@voltuer Did you intend to comment on a different issue? The `deno bundle` command was removed in Deno 2 and doesn't exist there.
